### PR TITLE
Remove Network Access in Tests

### PR DESCRIPTION
**Pull request by jamesvandyne**
This PR removes network access in the unit tests so our tests run faster, offline, and aren't subject to breaking when a third-party modifies their site. 

Fist I downloaded the content for each of the 5 network calls and stored them in our `tests/fixtures/` directory. Then I added their contents for each call using `responses`.

All tests now complete is less than a second.

### CODE REVIEW

File: src/indieweb_utils/replies/context.py [hunk 1]
```diff
@@ -75,11 +77,10 @@ def _process_h_entry_author(h_entry: dict, url: str, domain: str) -> Tuple[str,
         try:
             author = mf2py.parse(requests.get(author_url, timeout=10, verify=False).text)
         except requests.exceptions.RequestException:
-            author = {}
-
-        h_card = [item for item in author["items"] if item.get("type", []) == ["h-card"]][0]
-
-        author_url, author_name, author_image = _get_author_properties(author_url, h_card)
+            pass
```
Comment (jamesvandyne) [inline src/indieweb_utils/replies/context.py:80]: In the case of a network error, the previous code would result in a KeyError ( the `author` dict is empty and has no key `"items"`.  This changes the flow so we only over call `_get_author_properties` if we don't have a network error.

File: src/indieweb_utils/replies/context.py [hunk 2]
```diff
@@ -60,6 +60,8 @@ def _process_h_entry_author(h_entry: dict, url: str, domain: str) -> Tuple[str,
     parsed_url = url_parse.urlsplit(url)
 
     author_url = url
+    author_image = ""
+    author_name = ""
```
Comment (jamesvandyne) [inline src/indieweb_utils/replies/context.py:64]: Provide default values so we don't run into Undefined Variable errors.

File: tests/test_replies/test_context.py [hunk 3]
```diff
@@ -16,32 +23,43 @@ def test_reply_context_1(self):
...
-        assert (
-            reply_context.post_url
-            == "https://www.theguardian.com/technology/2022/jan/31/beats-fit-pro-review-apple-workout-ready-airpods-pro-rivals-battery-price"  # noqa
-        )
+        assert reply_context.post_url == url
         assert reply_context.webmention_endpoint == ""
         assert reply_context.authors[0].url == "https://www.theguardian.com"
         assert reply_context.authors[0].name == ""
-        assert reply_context.authors[0].photo == "https://static.guim.co.uk/images/favicon-32x32.ico"
+        assert reply_context.authors[0].photo == ""
```
Comment (jamesvandyne) [inline tests/test_replies/test_context.py:62]: It seems that their html must have changed as the favicon was no longer being set for the author url.

**Comment by capjamesg** [comment]
Looks good to me! I have merged this into the main branch. I shall begin work on preparing the next release.

**Pull request merged**